src/devices/Mcp3xxx/Mcp3Base.cs

```diff
@@ -0,0 +1,169 @@
+    // ==== ======== ========
+    // Req  xxxxxxxx xxxxxxxx
+    // Resp xxNRRRR RRRRRRRRx
+    //
+    // MCP3202
+    // Byte        0        1        2
+    // ==== ======== ======== ========
+    // Req  00000001 MC1xxxxx xxxxxxxx
+    // Resp xxxxxxxx xxxNRRRR RRRRRRRR
+    //
+    // MCP3204
+    // Byte        0        1        2
+    // ==== ======== ======== ========
+    // Req  00000SMx CCxxxxxx xxxxxxxx
+    // Resp xxxxxxxx xxDNRRRR RRRRRRRR
+    //
+    // MCP3208
+    // Byte        0        1        2
+    // ==== ======== ======== ========
+    // Req  00000SMC CCxxxxxx xxxxxxxx
+    // Resp xxxxxxxx xxDNRRRR RRRRRRRR
+    //
+    // MCP3301
+    // Byte        0        1
+    // ==== ======== ========
+    // Req  xxxxxxxx xxxxxxxx
+    // Resp xxN-RRRR RRRRRRRR
+    //
+    // MCP3302
+    // Byte        0        1        2
+    // ==== ======== ======== ========
+    // Req  0000SMxC Cxxxxxxx xxxxxxxx
+    // Resp xxxxxxxx xDN-RRRR RRRRRRRR
+    //
+    // MCP3304
+    // Byte        0        1        2
+    // ==== ======== ======== ========
+    // Req  0000SMCC Cxxxxxxx xxxxxxxx
+    // Resp xxxxxxxx xDN-RRRR RRRRRRRR
+    //
+    // S = StartBit = 1
+    // C = Channel
+    // M = SingleEnded
+    // D = Delay
+    // N = Null Bit = 0
+    // R = Response
+    // - = Sign Bit
+    // x = Dont Care
+    //
+
+    /// <summary>
+    /// Mcp3Base Abstract class representing the MCP ADC devices.
+    /// </summary>
+    public abstract class Mcp3Base : IDisposable
+    {
+        /// <summary>
+        /// InputType: the type of pin connection
+        /// </summary>
+        protected enum InputType
+        {
+            ///<summary>The value is measured as the voltage on a single pin</summary>
+            SingleEnded = 0,
+            ///<summary>The value is the difference in voltage between two pins with the first pin being the positive one</summary>
+            Differential = 1,
+            ///<summary>The value is the difference in voltage between two pins with the second pin being the positive one</summary>
+            InvertedDifferential = 2
+        }
+
+        private SpiDevice _spiDevice;
+
+        /// <summary>
+        /// Constructs Mcp3Base instance
+        /// </summary>
+        /// <param name="spiDevice">Device used for SPI communication</param>
+        public Mcp3Base(SpiDevice spiDevice)
+        {
+            _spiDevice = spiDevice;
+        }
+
+        /// <summary>
+        /// Disposes Mcp3Base instances
+        /// </summary>
+        public void Dispose()
+        {
+            _spiDevice?.Dispose();
+            _spiDevice = null;
+        }
+
+        /// <summary>
+        /// Reads a value from the device
+        /// </summary>
+        /// <param name="adcRequest">A bit pattern to be sent to the ADC.</param>
+        /// <param name="adcRequestLengthBytes">The number of bytes to be sent to the ADC containing the request and returning the response.</param>
```

Comment:
can this be calculated with adcResolutionBits? i.e.: `((adcResolutionBits + 3) / 8)` or something along these lines

Reply:
I have made a change so that it looks for a start bit in bits 16-23 and if it is found then the buffer size is 3 bytes otherwise 2.